configs/src/networks/types.ts

```diff
@@ -0,0 +1,37 @@
+export enum SupportedAliases {
+  ETHERS_5 = 'ethers5',
+  ETHERS_6 = 'ethers6',
+  ALCHEMY_SUBGRAPHS = 'alchemySubgraphs',
+}
+
+export enum SupportedNetworks {
+  MAINNET = 'mainnet',
+  GOERLI = 'goerli',
+  SEPOLIA = 'sepolia',
+  POLYGON = 'polygon',
+  MUMBAI = 'mumbai',
+  BASE = 'baseMainnet',
+  BASE_GOERLI = 'baseGoerli',
+  BASE_SEPOLIA = 'baseSepolia',
+  ARBITRUM = 'arbitrum',
+  ARBITRUM_SEPOLIA = 'arbitrumSepolia',
+  LOCAL = 'local',
+}
+
+export type NetworkConfig = {
+  url: string;
+  isTestnet: boolean;
+  chainId: number;
+  name: SupportedNetworks;
+  feesUrl?: string;
+  gasPrice?: number;
+  aliases: NetworkAliases;
+};
+
+export type NetworkAliases = {
+  [index in SupportedAliases]?: string;
+};
+
+export type NetworkConfigs<T = NetworkConfig> = {
```

Comment:
Why have the generic in here out of interest? It seems like the type is perfectly captured by the NetworkConfig. Just curious in what situation would you expect to swap it out?